crates/assists/src/handlers/replace_impl_trait_with_generic.rs

```diff
@@ -0,0 +1,62 @@
+use syntax::ast::{self, edit::AstNodeEdit, make, AstNode, GenericParamsOwner};
+
+use crate::{AssistContext, AssistId, AssistKind, Assists};
+
+// Assist: replace_impl_trait_with_generic
+//
+// Replaces `impl Trait` function argument with the named generic.
+pub(crate) fn replace_impl_trait_with_generic(
+    acc: &mut Assists,
+    ctx: &AssistContext,
+) -> Option<()> {
+    let type_impl_trait = ctx.find_node_at_offset::<ast::ImplTraitType>()?;
+    let type_param = type_impl_trait.syntax().parent().and_then(ast::Param::cast)?;
+    let type_fn = type_param.syntax().ancestors().nth(2).and_then(ast::Fn::cast)?;
+
+    let generic_param_list =
+        type_fn.generic_param_list().unwrap_or_else(|| make::generic_param_list(None));
```

Comment:
Microoptimization: we can move this inside the close, so that it only is avaulated when we actually apply the assist.